plugin/pkg/scheduler/api/types.go

```diff
@@ -100,3 +108,61 @@ type LabelPreference struct {
 	// If false, higher priority is given to nodes that do not have the label
 	Presence bool `json:"presence"`
 }
+
+// Holds the parameters used to communicate with the extender. If a verb is unspecified/empty,
+// it is assumed that the extender chose not to provide that extension.
+type ExtenderConfig struct {
+	// URLPrefix at which the extender is available
+	URLPrefix string `json:"urlPrefix"`
+	// Verb for the filter call, empty if not supported
+	FilterVerb string `json:"filterVerb,omitempty"`
+	// Verb for the prioritize call, empty if not supported
+	PrioritizeVerb string `json:"prioritizeVerb,omitempty"`
+	// The numeric multiplier for the node scores that the prioritize call generates.
+	// The weight should be a positive integer
+	Weight int `json:"weight,omitempty"`
+	// EnableHttps specifies whether https should be used to communicate with the extender
+	EnableHttps bool `json:"enableHttps,omitempty"`
+	// TLSConfig specifies the transport layer security config
+	TLSConfig *client.TLSClientConfig `json:"tlsConfig,omitempty"`
+	// HTTPTimeout specifies the timeout duration for a call to the extender. Filter timeout
+	// fails the scheduling of the pod. Prioritize timeout is ignored, k8s/other extenders
+	// priorities are used to select the node.
+	HTTPTimeout time.Duration `json:"httpTimeout,omitempty"`
+}
+
+// ExtenderArgs represents the arguments needed by the extender to filter/prioritize
+// nodes for a pod.
+type ExtenderArgs struct {
+	Pod   api.Pod      `json:"pod"`
+	Nodes api.NodeList `json:"nodes"`
+}
+
+// ExtenderFilterResult represents the results of a filter call to an extender
+type ExtenderFilterResult struct {
+	Nodes api.NodeList `json:"nodes,omitempty"`
+	Error string       `json:"error,omitempty"`
+}
+
+// HostPriority represents the priority of scheduling to a particular host, lower priority is better.
+type HostPriority struct {
+	Host  string `json:"host"`
+	Score int    `json:"score"`
+}
+
+type HostPriorityList []HostPriority
+
+func (h HostPriorityList) Len() int {
```

Comment:
I realize that the previous iteration of these didn't have Godoc either but they really should as they are exported.